token/program-2022/src/pod.rs

```diff
@@ -60,6 +73,30 @@ impl From<OptionalNonZeroPubkey> for COption<Pubkey> {
     }
 }
 
+#[cfg(feature = "serde-traits")]
+impl Serialize for OptionalNonZeroPubkey {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+    {
+        if self.0 == Pubkey::default() {
+            s.serialize_none()
+        } else {
+            s.serialize_some(&self.0.to_string())
+        }
+    }
+}
+
+#[cfg(feature = "serde-traits")]
+impl<'de> Deserialize<'de> for OptionalNonZeroPubkey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: Deserializer<'de>,
+    {
+        deserializer.deserialize_any(OptionalNonZeroPubkeyVisitor)
+    }
+}
```

Comment:
Is there a reason that the serde traits are implemented here rather than in `serialization.rs`? It seems like it would be cleaner to follow the same model as everywhere else by defining the type + visitor there and then using `serde(with = ...`

Reply:
I was trying to use `serde_with` only when working with external types since I cannot `impl Serialize`/`impl Deserialize` on them as they are imported from external crates. 

Another thing is that for `OptionalNonZeroPubkey` it's inner `Pubkey` is private which makes it harder to instantiate it from a separate module. Currently in it's `Visitor` in `serialization.rs` I am using  `OptionalNonZeroPubkey::try_from(Some(pkey))` to instantiate it.

If you prefer I can move the current serialization for `OptionalNonZeroPubkey`/`OptionalNonZeroElGamalPubkey` into `serialization.rs` and then use `serde(with = ...`, but if you could advise what is the best way to instantiate them since their inner `Pubkey` is private.

Reply:
Ah I see what you mean. Makes sense to me, thanks for the explanation!